node/src/protocol.rs

```diff
@@ -43,6 +46,29 @@ pub enum Message {
     FinalitySignature(Box<FinalitySignature>),
 }
 
+impl Payload for Message {
+    #[inline]
+    fn classify(&self) -> MessageKind {
+        match self {
+            Message::Consensus(_) => MessageKind::Consensus,
+            Message::DeployGossiper(_) => MessageKind::DeployGossip,
+            Message::AddressGossiper(_) => MessageKind::AddressGossip,
+            Message::GetRequest { tag, .. } | Message::GetResponse { tag, .. } => {
+                match tag {
+                    Tag::Deploy => MessageKind::DeployTransfer,
+                    Tag::Block => MessageKind::BlockTransfer,
+                    // This is a weird message, which we should not encounter here?
+                    Tag::GossipedAddress => MessageKind::Other,
+                    Tag::BlockByHeight => MessageKind::BlockTransfer,
+                    Tag::BlockHeaderByHash => MessageKind::BlockTransfer,
+                    Tag::BlockHeaderAndFinalitySignaturesByHeight => MessageKind::BlockTransfer,
+                }
+            }
+            Message::FinalitySignature(_) => MessageKind::BlockTransfer,
```

Comment:
We should disambiguate between `Message::FinalitySignature` and `Tag::BlockByHeight`, `Tag::BlockHeaderByHash` and `Tag::BlockHeaderAndFinalitySignaturesByHeight` – the latter are issues on joining node while the latter (`FinalitySignature(_)`) is created during normal node operation.

Reply:
I wanted to be fairly course and not explode the number of categories.

Should I move `FinalitySignature` into the `Consensus` class maybe?

Reply:
If we don't want to add more categories then `Consensus` is :+1: